polyfill/lib/ecmascript.mjs

```diff
@@ -743,14 +744,61 @@ export const ES = ObjectAssign(ObjectAssign({}, ES2019), {
     nanosecond = ES.ConstrainToRange(nanosecond, 0, 999);
     return { hour, minute, second, millisecond, microsecond, nanosecond };
   },
+  ConstrainDateTime: (year, month, day, hour, minute, second, millisecond, microsecond, nanosecond) => {
+    year = ES.ConstrainToRange(year, YEAR_MIN, YEAR_MAX);
+    month = ES.ConstrainToRange(month, 1, 12);
+    day = ES.ConstrainToRange(day, 1, ES.DaysInMonth(year, month));
+    ({
+      hour,
+      minute,
+      second,
+      millisecond,
+      microsecond,
+      nanosecond,
+    } = ES.ConstrainTime(hour, minute, second, millisecond, microsecond, nanosecond));
+    // Constrain to within 24 hours outside the Absolute range
+    if (year === YEAR_MIN) {
+      try {
+        ES.GetEpochFromParts(year, month, day + 1, hour, minute, second, millisecond, microsecond, nanosecond - 1);
+      } catch (e) {
+        if (!(e instanceof RangeError)) throw e;
+        month = 4;
+        day = 19;
+        hour = minute = second = millisecond = microsecond = 0;
+        nanosecond = 1;
+      }
```

Comment:
I feel like this implementation relying on the exception type is somewhat brittle; could you review for alternatives?

Reply:
Err, of course -1 is not a good out of range value :face_with_head_bandage:
`null` maybe?